src/main/java/org/folio/search/service/reindex/jdbc/CallNumberRepository.java

```diff
@@ -0,0 +1,184 @@
+
+    jdbcTemplate.batchUpdate(instanceCallNumberSql, entityBatch.relationshipEntities(), BATCH_OPERATION_SIZE,
+      (statement, entity) -> {
+        statement.setString(1, getCallNumberId(entity));
+        statement.setString(2, getItemId(entity));
+        statement.setString(3, getInstanceId(entity));
+        statement.setString(4, getTenantId(entity));
+        statement.setString(5, getLocationId(entity));
+      });
+  }
+
+  private void saveResourceEntities(ChildResourceEntityBatch entityBatch) {
+    var callNumberTable = getFullTableName(context, entityTable());
+    var callNumberSql = INSERT_ENTITIES_SQL.formatted(callNumberTable);
+
+    jdbcTemplate.batchUpdate(callNumberSql, entityBatch.resourceEntities(), BATCH_OPERATION_SIZE,
+      (statement, entity) -> {
+        statement.setString(1, getId(entity));
+        statement.setString(2, getCallNumber(entity));
+        statement.setString(3, getPrefix(entity));
+        statement.setString(4, getSuffix(entity));
+        statement.setString(5, getTypeId(entity));
+        statement.setString(6, getVolume(entity));
+        statement.setString(7, getEnumeration(entity));
+        statement.setString(8, getChronology(entity));
+        statement.setString(9, getCopyNumber(entity));
+      });
+  }
+
+  private String getCallNumber(Map<String, Object> callNumberComponents) {
+    return getString(callNumberComponents, "callNumber");
+  }
+
+  private String getCallNumberId(Map<String, Object> callNumberComponents) {
+    return getString(callNumberComponents, "callNumberId");
+  }
+
+  private String getLocationId(Map<String, Object> item) {
+    return getString(item, "locationId");
+  }
+
+  private String getTenantId(Map<String, Object> item) {
+    return getString(item, "tenantId");
+  }
+
+  private String getInstanceId(Map<String, Object> item) {
+    return getString(item, "instanceId");
+  }
+
+  private String getItemId(Map<String, Object> item) {
+    return getString(item, "itemId");
+  }
+
+  private String getId(Map<String, Object> item) {
+    return getString(item, "id");
+  }
+
+  private String getCopyNumber(Map<String, Object> item) {
+    return getString(item, "copyNumber");
+  }
+
+  private String getEnumeration(Map<String, Object> item) {
+    return getString(item, "enumeration");
+  }
+
+  private String getChronology(Map<String, Object> item) {
+    return getString(item, "chronology");
+  }
+
+  private String getVolume(Map<String, Object> item) {
+    return getString(item, "volume");
+  }
+
+  private String getTypeId(Map<String, Object> callNumberComponents) {
+    return getString(callNumberComponents, "callNumberTypeId");
+  }
+
+  private String getSuffix(Map<String, Object> callNumberComponents) {
+    return getString(callNumberComponents, "callNumberSuffix");
+  }
+
+  private String getPrefix(Map<String, Object> callNumberComponents) {
+    return getString(callNumberComponents, "callNumberPrefix");
+  }
```

Comment:
extracting specific call number entity field from a map is also needed in other places I guess, like in CallNumberResourceExtractor, so maybe makes sense to unify them into some place and make it accessible.

Reply:
There is different maps in repository and in extractor, so methods will be different. I think we should refactor all this extractors and repositories to work with pojos and not maps.